libass/ass.c

```diff
@@ -264,14 +264,24 @@ static long long string2timecode(ASS_Library *library, char *p)
 
 #define STRVAL(name) \
     } else if (ass_strcasecmp(tname, #name) == 0) { \
-        if (target->name != NULL) free(target->name); \
-        target->name = strdup(token);
+        char* new_str = strdup(token); \
+        if (new_str) { \
+            free(target->name); \
+            target->name = new_str; \
+        } else { \
+            ass_nomem_log(track->library, true, "Failed to strdup field '" #name "'"); \
```

Comment:
IIUC, in this case, the old value will exist only if this is a user override. `NULL` isn’t valid here, so the choice is either skipping the failing override or discarding the style entirely.

Reply:
Yes, as atiob said, in initial parsing the previous value is NULL and if nothing valid was read the event/style will be discarded directly after parsing.
In a previous version, the NULL check was exclusive to style overrides, but as suggested by MrSmile I merged the check into the regular variant: https://github.com/libass/libass/pull/411#discussion_r453262770
In v5 I also added the nomem_log-reporting.

Reply:
Most parsing-related nomem_logs have been moved to the caller with v6.
However, in case of style overrides this doesn't easily work. So either:
 - string macros keep always reporting failures
 - style override stops using the string macros and uses unique semi-duplicated code again similar to [a previous version](https://github.com/TheOneric/libass/commit/8e905c03f6dd8e4a8735f72bf8c7c4378a1833ca) (in which case the strdup check can also be removed from the macro)

I'd be fine with either option; what are your preferences?

Reply:
I think the main problem was when errors were _both_ logged _and_ returned. String parsing macros don’t return errors, so logging them in place is fine.